pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -86,9 +87,13 @@ type Role string
 
 // K8SSelectorConfig is Kubernetes Selector Config
 type K8SSelectorConfig struct {
-	// +kubebuilder:validation:Required
-	Role  Role   `json:"role"`
+	// +required
+	Role Role `json:"role"`
+	// +kubebuilder:validation:MinLength=1
+	// +optional
 	Label string `json:"label,omitempty"`
```

Comment:
you need a string pointer if you mean optional but a lenght of at least 1 if defined.